### PR TITLE
Add Ability to Disable Replication Status Endpoints in Listener Configuration

### PR DESCRIPTION
**Pull request by marcboudreau**
This PR adds the ability to parse a configuration setting in the TCP listener configuration section so that the value can be stored in the http.Request context. This will allow the enterprise version implementation of the replication status endpoints:
- `sys/replication/status`
- `sys/replication/dr/status`
- `sys/replication/performance/status`
to react accordingly.

### CODE REVIEW

File: website/content/docs/configuration/listener/tcp.mdx [hunk 1]
```diff
@@ -202,6 +202,10 @@ default value in the `"/sys/config/ui"` [API endpoint](/vault/api-docs/system/co
   there is no X-Forwarded-For header or it is empty, the client address will be
   used as-is, rather than the client connection rejected.
 
+- `disable_replication_status_endpoints` `(bool: false)` - If set true, the
+  replication status endpoints will be disabled for requests sent to this
+  listener.
```
Comment (schavis) [inline website/content/docs/configuration/listener/tcp.mdx:207]:
```suggestion
- `disable_replication_status_endpoints` `(bool: false)` - Disables replication
  status endpoints for the configured listener when set to `true`.
```

Style correction: write in active voice, avoid "this" as a pronoun

**Review by schavis — CHANGES_REQUESTED** [review]

File: http/util.go [hunk 2]
```diff
@@ -133,6 +133,14 @@ func rateLimitQuotaWrapping(handler http.Handler, core *vault.Core) http.Handler
 	})
 }
 
+func disableReplicationStatusEndpointWrapping(h http.Handler) http.Handler {
```
Comment (peteski22) [inline http/util.go:136]: Just needs a go doc comment here 😄

File: website/content/docs/configuration/listener/tcp.mdx (hunk 1 above)
Comment (marcboudreau) [reply website/content/docs/configuration/listener/tcp.mdx:207]: Thanks, this reads much better.

**Comment by marcboudreau** [comment]
@schavis, I've make the edit you suggested.

**Review by schavis — APPROVED** [review]

**Pull request merged**